### ISSUE TITLE
🐞 Bug report: Console error when using global widget - [ED-10226]

### ISSUE THREAD

**Issue by sarailoo** [elementor/elementor#19456]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

When I try to use global widget I have this console error:
editor-document.min.js?ver=3.7.0:2 Uncaught ReferenceError: elementorDevTools is not defined
    at get CommandHistory [as CommandHistory] (editor-document.min.js?ver=3.7.0:2:25976)
    at 43670 (editor.min.js?ver=3.7.3:3:61401)
    at __webpack_require__ (editor.min.js?ver=3.7.3:3:276153)
    at 36112 (editor.min.js?ver=3.7.3:3:59189)
    at __webpack_require__ (editor.min.js?ver=3.7.3:3:276153)
    at 51311 (editor.min.js?ver=3.7.3:3:64197)
    at __webpack_require__ (editor.min.js?ver=3.7.3:3:276153)
    at 12044 (editor.min.js?ver=3.7.3:3:83262)
    at __webpack_require__ (editor.min.js?ver=3.7.3:3:276153)
    at editor.min.js?ver=3.7.3:3:279308

For fixing the issue, you have to replace $e.modules.document.CommandHistory with $e.modules.editor.document.CommandHistoryBase in ../modules/global-widget/assets/js/editor/commands/unlink.js and ../modules/global-widget/assets/js/editor/commands/link.js because the CommandHistory is deprecated in Elementor 3.7.0

### Steps to reproduce

- Add a new page
- Open console
- Edit with Elementor
- You can see the error with elementor 3.7.0 and Elementor Pro 3.7.3

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

<details>
<summary>Click to reveal</summary>

````txt



== Server Environment ==
	Operating System: Linux
	Software: Apache/2.4.53 (Debian)
	MySQL version: Source distribution v10.3.34
	PHP Version: 7.4.29
	PHP Memory Limit: 500M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 500M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.0.1
	Site URL: http://localhost:8055
	Home URL: http://localhost:8055
	WP Multisite: No
	Max Upload Size: 500 MB
	Memory limit: 40M
	Max Memory limit: 500M
	Permalink Structure: Plain
	Language: en-US
	Timezone: 0
	Debug Mode: Active

== Theme ==
	Name: Twenty Twenty-One
	Version: 1.6
	Author: the WordPress team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36

== Active Plugins ==
	Advanced Custom Fields
		Version: 5.12.3
		Author: Delicious Brains

	Awesome Support
		Version: 6.1.0
		Author: Awesome Support Team

	Custom Post Type UI
		Version: 1.12.1
		Author: WebDevStudios

	Elementor
		Version: 3.7.0
		Author: Elementor.com

	Elementor Pro
		Version: 3.7.3
		Author: Elementor.com

	JetElements For Elementor
		Version: 2.6.6
		Author: Crocoblock

	JetMenu
		Version: 2.1.7
		Author: Crocoblock

	JetPopup
		Version: 1.5.6
		Author: Crocoblock

	Meta Box
		Version: 5.6.6
		Author: MetaBox.io

	Meta Box - Elementor Integrator
		Version: 2.1.1
		Author: MetaBox.io

	Product Import Export for WooCommerce
		Version: 2.2.2
		Author: WebToffee

	Sellkit
		Version: 1.2.5
		Author: Artbees

	Sellkit Pro
		Version: 1.2.4
		Author: Artbees

	SumUp Payment Gateway For WooCommerce
		Version: 1.0.0
		Author: SumUp

	UpdraftPlus - Backup/Restore
		Version: 1.22.12
		Author: UpdraftPlus.Com, DavidAnderson

	WebToffee PayPal Express Checkout Payment Gateway for WooCommerce ( Free )
		Version: 1.7.4
		Author: WebToffee

	WooCommerce
		Version: 6.6.1
		Author: Automattic

	WP All Export
		Version: 1.3.7
		Author: Soflyy

	WP Cost Estimation &amp; Payment Forms Builder
		Version: 10.1.28
		Author: Biscay Charly (loopus)

	WunderWP
		Version: 1.6.1
		Author: Artbees

	YITH Frontend Manager for WooCommerce Premium
		Version: 1.6.21
		Author: YITH

	YITH Point of Sale for WooCommerce
		Version: 2.1.3-beta.1
		Author: YITH

	YITH Point of Sale for WooCommerce
		Version: 1.0.5
		Author: YITH

	YITH WooCommerce Dynamic Pricing and Discounts Premium
		Version: 3.5.0
		Author: YITH

	YITH WooCommerce Multi Vendor Premium
		Version: 3.15.0
		Author: YITH

	YITH WooCommerce Product Add-ons &amp; Extra Options Premium
		Version: 3.4.1
		Author: YITH

	YITH WooCommerce Product Bundles
		Version: 1.12.0
		Author: YITH

	YITH WooCommerce Request A Quote Premium
		Version: 4.3.0
		Author: YITH

	ZMI Core
		Version: 1.0
		Author: WPDone


== Must-Use Plugins ==
	Elementor Safe Mode
		Version: 1.0.0
		Author: Elementor.com


== Elementor Experiments ==
	Optimized DOM Output: Active by default
	Improved Asset Loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Inactive by default
	Accessibility Improvements: Active by default
	Additional Custom Breakpoints: Active by default
	Import Export Template Kit: Active by default
	Hide native WordPress widgets from search results: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Inactive by default
	Default to New Theme Builder: Active by default
	Landing Pages: Active by default
	Color Sampler: Active by default
	Favorite Widgets: Active by default
	Admin Top Bar: Active by default
	Page Transitions: Active by default
	Notes: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default


== Log ==
	
Log: showing 6 of 62022-07-04 08:14:47 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.6.6',
  'to' => '3.6.7',
)]
2022-08-12 12:27:27 [info] elementor::elementor_updater Started 
2022-08-12 12:27:27 [info] Elementor/Upgrades - _on_each_version Start  
2022-08-12 12:27:27 [info] Elementor/Upgrades - _on_each_version Finished 
2022-08-12 12:27:27 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.6.6',
  'to' => '3.7.0',
)]
2022-08-12 14:12:26 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.7.0',
  'to' => '3.7.3',
)]

PHP: showing 12 of 12PHP: 2022-07-11 07:20:33 [notice X 6][/var/www/html/wp-content/plugins/elementor/core/common/modules/ajax/module.php::171] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-07-14 11:22:01 [warning X 1][/var/www/html/wp-content/plugins/elementor/core/editor/editor.php::116] Cannot modify header information - headers already sent by (output started at /var/www/html/wp-content/plugins/jupiterx-core/includes/control-panel-2/class.php:354) [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-07-14 17:29:50 [warning X 1][/var/www/html/wp-content/plugins/elementor/includes/db.php::267] Invalid argument supplied for foreach() [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-07-14 18:41:08 [notice X 1][/var/www/html/wp-content/plugins/elementor/includes/template-library/sources/local.php::1370] Undefined index: content [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-07-14 18:42:37 [notice X 4][/var/www/html/wp-content/plugins/elementor/includes/template-library/sources/local.php::1396] Undefined index: type [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-07-18 09:20:00 [notice X 4][/var/www/html/wp-content/plugins/elementor/core/app/modules/import-export/module.php::90] Trying to get property &#039;label&#039; of non-object [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-07-29 00:19:26 [notice X 2][/var/www/html/wp-content/plugins/elementor-pro/license/updater.php::83] Undefined index: new_version [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-07-29 00:19:31 [error X 3][/var/www/html/wp-content/plugins/elementor-pro2/elementor-pro.php::75] Cannot redeclare elementor_pro_load_plugin() (previously declared in /var/www/html/wp-content/plugins/elementor-pro/elementor-pro.php:61) [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-07-29 08:30:16 [error X 10][/var/www/html/wp-content/plugins/elementor/includes/base/controls-stack.php::377] Uncaught ArgumentCountError: Too few arguments to function Elementor\Controls_Stack::add_control(), 0 passed in /var/www/html/wp-content/plugins/jupiterx-core/includes/extensions/raven/includes/core/document-types/types/popup.php on line 422 and at least 2 expected in /var/www/html/wp-content/plugins/elementor/includes/base/controls-stack.php:377
Stack trace:
#0 /var/www/html/wp-content/plugins/jupiterx-core/includes/extensions/raven/includes/core/document-types/types/popup.php(422): Elementor\Controls_Stack-&gt;add_control()
#1 /var/www/html/wp-content/plugins/elementor/includes/base/controls-stack.php(2282): JupiterX_Core\Raven\Core\Document_Types\Type\Jupiterx_Popup_Document-&gt;register_controls()
#2 /var/www/html/wp-content/plugins/elementor/includes/base/controls-stack.php(494): Elementor\Controls_Stack-&gt;init_controls()
#3 /var/www/html/wp-content/plugins/elementor/includes/base/controls-stack.php(300): Elementor\Controls_Stack-&gt;get_stack()
#4 /var/www/html/wp-content/plugins/elementor/core/settings/page/mode [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-07-29 14:53:07 [notice X 9][/var/www/html/wp-content/plugins/elementor/includes/controls/base-multiple.php::87] Undefined index: size [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-08-01 15:43:58 [warning X 1][/var/www/html/wp-content/plugins/elementor-pro/plugin-update-checker/Puc/v4p6/UpdateChecker.php::400] The URL https://dl.elementor-site.ir/json/elementorpro-321.json?installed_version=3.7.0&amp;php=7.4.29&amp;locale=en_US&amp;checking_for_updates=1 does not point to a valid metadata file. WP HTTP Error: cURL error 28: Operation timed out after 10000 milliseconds with 0 out of 0 bytes received [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-08-04 21:55:04 [error X 2][/var/www/html/wp-content/plugins/elementor-pro/modules/posts/widgets/posts.php::47] syntax error, unexpected &#039;$this&#039; (T_VARIABLE) [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 3 of 3JS: 2022-08-12 13:37:59 [error X 45][http://localhost:8055/wp-content/plugins/elementor/assets/js/editor-document.min.js?ver=3.7.0:2:25976] elementorDevTools is not defined 
JS: 2022-08-12 19:08:09 [error X 2][http://localhost:8055/wp-content/plugins/elementor/assets/js/common-modules.min.js?ver=3.7.0:2:51477] n.isValidChild() should be implemented, please provide \&#039;isValidChild\&#039; functionality. 
JS: 2022-08-12 19:27:18 [error X 3][http://localhost:8055/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:17421] Could not find template: \&quot;#tmpl-raven-panel-global-widget-no-templates\&quot; 



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatibility not specified
	JetElements For Elementor: Compatibility not specified
	Meta Box - Elementor Integrator: Compatibility not specified

== Elementor Pro - Compatibility Tag ==
	


````

</details>

**Comment by Stas0238** [comment]
+1

**Comment by Stas0238** [comment]
Hello,

It seems like someone in Elementor Dev team wanted to optimize the code but missed that such variable could be undefined 
https://prnt.sc/OUFzwHj6icQ4
Old version had -> elementorCommon.helpers
New version has -> elementorDevTools.deprecation 

Also I found that in previous versions there were no dev-tools (js file) which already exists in latest one.

For Elementor team I also found a possible case of what could be the reason of such issues for my customers and many more your Elementor's clients (if I found such issue even on github that makes sense it is big global issue of Elementor latest update )

Maybe the main problem is in initialization of dev-tools (module) because as I see how there was initialized elementorCommon there were no additional conditions and it was initialized in any case. (Just my opinion) -> https://prnt.sc/gyt4PorW_CZo

P.S. waiting for your fixupdate ASAP because we cannot fix it from our side and every day we have +few clients who write us with such issue. (for this time we recommend them to downgrade versions of plugin)

Regards, @Stas0238

**Comment by nufrankz** [comment]
+1.

**Comment by jannnfe** [comment]
+1

**Comment by hemanttejwani2610** [comment]
+1

**Comment by sirhenryhu** [comment]
@Stas0238 , thank you for bringing this issue up again. In fact, I do remember this issue from another topic and it seemed to be resolved and somehow it is back again. I am not familiar with coding at all but I do realise that Elementor global widget now stops working properly, which is pretty frustrating that I now have to go into different pages and posts to update the code so it would not interrupt the customer journey. I Hope Elementor team doesn't sleep as usual and takes this issue seriously....

**Comment by rf-999** [comment]
+1

**Comment by Ping123** [comment]
+1

**Comment by reinhardjung** [comment]
+1 :-(

**Comment by nicholaszein** [comment]
Hello everyone!

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro v3.19.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
